File: bro-compiler/src/main/java/me/xx2bab/bro/compiler/classloader/GradleClassLoader.java
package me.xx2bab.bro.compiler.classloader;

import java.net.MalformedURLException;
import java.net.URL;
import java.net.URLClassLoader;
import java.util.ArrayList;
import java.util.List;

public class GradleClassLoader {

    private URLClassLoader urlClassLoader;

    public GradleClassLoader(String[] classPaths) {
        List<URL> urls = new ArrayList<>();
        for(String classPath : classPaths) {
            try {
                urls.add(new URL(classPath));
            } catch (MalformedURLException e) {
                throw new IllegalArgumentException("Classloader URL " + classPath
                        + "is invalid.");
            }
        }
        urlClassLoader = new URLClassLoader(urls.toArray(new URL[0]), this.getClass().getClassLoader());
    }

    public Class load(String clazz) throws ClassNotFoundException {
        return urlClassLoader.loadClass(clazz);
    }

}

File: bro-compiler/src/main/java/me/xx2bab/bro/compiler/classloader/GradleClassLoader.kt
package me.xx2bab.bro.compiler.classloader

import java.net.MalformedURLException
import java.net.URL
import java.net.URLClassLoader
import java.util.*

class GradleClassLoader(classPaths: Array<String>) {

    private val urlClassLoader: URLClassLoader

    @Throws(ClassNotFoundException::class)
    fun load(clazz: String?): Class<*> {
        return urlClassLoader.loadClass(clazz)
    }

    init {
        val urls: MutableList<URL> = ArrayList()
        for (classPath in classPaths) {
            try {
                urls.add(URL(classPath))
            } catch (e: MalformedURLException) {
                throw IllegalArgumentException("Classloader URL " + classPath
                        + "is invalid.")
            }
        }
        urlClassLoader = URLClassLoader(urls.toTypedArray(), this.javaClass.classLoader)
    }
}